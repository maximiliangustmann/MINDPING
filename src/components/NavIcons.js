import React from 'react'
import styled from 'styled-components/macro'
import { ReactComponent as Book } from '../img/book.svg'
import { ReactComponent as Audiobook } from '../img/audiobook.svg'
import { ReactComponent as Podcast } from '../img/podcast.svg'
import { ReactComponent as Film } from '../img/film.svg'
import { ReactComponent as Series } from '../img/series.svg'

export default function NavIcons() {
  return (
    <IconWrapper>
      <BookIcon />
      <AudiobookIcon />
      <PodcastIcon />
      <FilmIcon />
      <SeriesIcon />
    </IconWrapper>
  )
}

const IconWrapper = styled.div`
  display: grid;
  grid-template-columns: 0.1fr 0.1fr 0.1fr 0.1fr 0.1fr;
  grid-template-rows: 1fr 1fr 1fr;
  row-gap: 20%;
  fill: var(--main-red);
  justify-content: center;
`

const BookIcon = styled(Book)`
  height: 80%;
  width: 80%;
  grid-column: 3;
  grid-row: 1;
`

const AudiobookIcon = styled(Audiobook)`
  height: 80%;
  width: 80%;
  grid-column: 1;
  grid-row: 2;
`

const PodcastIcon = styled(Podcast)`
  height: 80%;
  width: 80%;
  grid-column: 5;
  grid-row: 2;
`

const FilmIcon = styled(Film)`
  height: 80%;
  width: 80%;
  grid-column: 2;
  grid-row: 3;
`

const SeriesIcon = styled(Series)`
  height: 80%;
  width: 80%;
  grid-column: 4;
  grid-row: 3;
`
