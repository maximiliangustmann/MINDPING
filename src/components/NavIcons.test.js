import React from 'react'
import renderer from 'react-test-renderer'
import NavIcons from './NavIcons'

describe('NavIcons', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NavIcons />)
    expect(tree).toMatchSnapshot()
  })
})
