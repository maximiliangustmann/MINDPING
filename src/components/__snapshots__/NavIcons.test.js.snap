// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavIcons renders correctly 1`] = `
<div
  className="NavIcons__IconWrapper-sc-16s2ff9-0 kxMxGQ"
>
  <div
    className="NavIcons__BookIconWrapper-sc-16s2ff9-1 grYUZe"
  >
    <svg
      className="NavIcons__BookIcon-sc-16s2ff9-2 jrxLuF"
      data-cy="book-icon"
      onClick={[Function]}
    >
      book.svg
    </svg>
    <p>
      Books
    </p>
  </div>
  <div
    className="NavIcons__AudiobookIconWrapper-sc-16s2ff9-3 ezxPVz"
  >
    <svg
      className="NavIcons__AudiobookIcon-sc-16s2ff9-4 hccpMh"
      data-cy="audiobook-icon"
      onClick={[Function]}
    >
      audiobook.svg
    </svg>
    <p>
      Audiobooks
    </p>
  </div>
  <div
    className="NavIcons__PodcastIconWrapper-sc-16s2ff9-5 kWoZHs"
  >
    <svg
      className="NavIcons__PodcastIcon-sc-16s2ff9-6 bGZwLi"
      data-cy="podcast-icon"
      onClick={[Function]}
    >
      podcast.svg
    </svg>
    <p>
      Podcasts
    </p>
  </div>
  <div
    className="NavIcons__FilmIconWrapper-sc-16s2ff9-7 fTptlw"
  >
    <svg
      className="NavIcons__FilmIcon-sc-16s2ff9-8 ebRclw"
      data-cy="film-icon"
      onClick={[Function]}
    >
      film.svg
    </svg>
    <p>
      Films
    </p>
  </div>
  <div
    className="NavIcons__SeriesIconWrapper-sc-16s2ff9-9 eqLxXB"
  >
    <svg
      className="NavIcons__SeriesIcon-sc-16s2ff9-10 eswIuU"
      data-cy="series-icon"
      onClick={[Function]}
    >
      series.svg
    </svg>
    <p>
      Series
    </p>
  </div>
</div>
`;
